openlibrary/plugins/openlibrary/code.py

```diff
@@ -814,6 +815,25 @@ def GET(self):
         h = guppy.hpy()
         return delegate.RawText(str(h.heap()))
 
+def _get_component(workid):
+    work = web.ctx.site.get('/works/%s' % workid) or {}
+    component = render_template('books/RelatedWorksCarousel', work)
+    return {0: str(component)}
+
+def get_cached_component(*args, **kwargs):
+    memoized_get_component_metadata = cache.memcache_memoize(
+        _get_component, "book.bookspage.component", timeout=dateutil.HALF_DAY_SECS)
+    return (memoized_get_component_metadata(*args, **kwargs) or
+            memoized_get_component_metadata.update(*args, **kwargs)[0])
+
+class Partials(delegate.page):
+    path = '/partials'
```

Comment:
Suggestion:

Let's make sure Partials is generic and that the correct _get_component function is called according to the parameters provided.

What we probably want to happen is within Partials.GET... We can pull the component out from our GET parameters as `component = i.pop("_component")` and then we call the corresponding function. `locals(component)(**i)` will call the correct function (whose value is represented by the component string) and then this will pass in to that function the remaining parameters specific by `i`.

e.g. {"work_id": "OL123W", "component": "related_work_carousels"} -->
related_work_carousels(**i)
related_work_carousel(work_id="OL123W")